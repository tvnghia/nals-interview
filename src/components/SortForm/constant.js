export const SORT_RADIOS = Object.freeze([
  {
    label: 'Ascending',
    value: 'asc'
  },

  {
    label: 'Descending',
    value: 'desc'
  },

  {
    label: 'None',
    value: 'none'
  }
])
