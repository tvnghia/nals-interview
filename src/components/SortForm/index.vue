<template>
  <div class="sort-form u-mt-16">
    <p class="sort-form__title">Sort blogs:</p>

    <div v-for="option in sortOption" :key="option.index" class="u-flex u-items-center">
      <input type="radio" name="sort_blogs" :id="option.id" :value="option.value" v-model="picked" />

      <label :for="option.id">{{ option.label }}</label>
    </div>
  </div>
</template>

<script>
import { SORT_RADIOS } from './constant'

export default {
  name: 'SortForm',

  props: {
    value: {
      type: String,
      default: 'asc'
    }
  },

  data() {
    return {
      sortOption: SORT_RADIOS,
      picked: this.value
    }
  },

  watch: {
    picked: {
      handler(value) {
        this.$emit('input', value)
      }
    }
  }
}
</script>

<style lang="scss" scoped>
.sort-form {
  font-size: 14px;

  &__title {
    font-size: 18px;
    font-weight: 500;
  }

  input {
    margin-right: 4px;
  }

  &,
  input,
  label {
    cursor: pointer;
  }
}
</style>
