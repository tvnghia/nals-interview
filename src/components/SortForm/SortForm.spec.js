import { shallowMount } from '@vue/test-utils'
import wrapperOps from '@/config-test/wrapper'
import SortForm from '@/components/SortForm'

describe('SortForm.vue', () => {
  let wrapper

  beforeEach(async () => {
    wrapper = shallowMount(SortForm, wrapperOps)
  })

  afterEach(() => {
    wrapper.vm.$destroy()
  })

  describe('when SortForm was mounted', () => {
    it('should SortForm render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when sort', () => {
    it('should emit sort value', async () => {
      await wrapper.setData({ picked: 'desc' })
      expect(wrapper.emitted('input')).toHaveLength(1)
      expect(wrapper.emitted('input')[0]).toEqual(['desc'])
    })
  })
})
