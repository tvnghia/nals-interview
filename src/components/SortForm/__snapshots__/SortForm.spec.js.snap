// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortForm.vue when SortForm was mounted should SortForm render correctly 1`] = `
<div class="sort-form u-mt-16">
  <p class="sort-form__title">Sort blogs:</p>
  <div class="u-flex u-items-center"><input type="radio" name="sort_blogs" id="1" value="asc"> <label for="1">Ascending</label></div>
  <div class="u-flex u-items-center"><input type="radio" name="sort_blogs" id="2" value="desc"> <label for="2">Descending</label></div>
</div>
`;
